phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CompactionScanner.java

```diff
@@ -84,45 +110,99 @@ public CompactionScanner(RegionCoprocessorEnvironment env,
+        LOGGER.info(String.format("CompactionScanner params:- (" +
+                        "physical-data-tablename = %s, compaction-tablename = %s, " +
+                        "emptyCF = %s, emptyCQ = %s, " +
+                        "minVersion = %d, maxVersion = %d, keepDeletedCells = %s, " +
+                        "familyCount = %d, localIndex = %s, emptyCFStore = %s, " +
+                        "compactionTime = %d, maxLookbackWindowStart = %d, maxLookbackInMillis = %d)",
+                table.getName().toString(), tableName,
+                Bytes.toString(this.emptyCF), Bytes.toString(emptyCQ),
+                this.minVersion, this.maxVersion, this.keepDeletedCells.name(),
+                this.familyCount, this.localIndex, this.emptyCFStore,
+                compactionTime, maxLookbackWindowStart, maxLookbackInMillis));
+
+    }
+
+    /**
+     * Helper method to create TTL tracker for various phoenix data model objects
+     * i.e views, view indexes ...
+     * @param env
+     * @param store
+     * @param baseTable
+     * @return
+     */
+    private TTLTracker createTTLTrackerFor(RegionCoprocessorEnvironment env,
+            Store store,  PTable baseTable) throws IOException {
+
+        long currentTime = EnvironmentEdgeManager.currentTimeMillis();
+        String compactionTableName = env.getRegion().getRegionInfo().getTable().getNameAsString();
+        String schemaName = SchemaUtil.getSchemaNameFromFullName(baseTable.getName().toString());
+        String tableName = SchemaUtil.getTableNameFromFullName(baseTable.getName().toString());
+
+        boolean isPartitionedIndexTable = false;
+        if (compactionTableName.startsWith(MetaDataUtil.VIEW_INDEX_TABLE_PREFIX)) {
+            isPartitionedIndexTable = true;
+        }
+
+        // NonPartitioned: Salt bucket property can be separately set for base tables and indexes.
+        // Partitioned: Salt bucket property can be set only for the base table.
+        // Global views, Tenant views, view indexes inherit the salt bucket property from their
+        // base table.
+        boolean isSalted = baseTable.getBucketNum() != null;
+        try (PhoenixConnection serverConnection = QueryUtil.getConnectionOnServer(new Properties(),
+                env.getConfiguration()).unwrap(PhoenixConnection.class)) {
+
+            Table childLinkHTable = serverConnection.getQueryServices().getTable(SYSTEM_CHILD_LINK_NAME_BYTES);
+            // If there is atleast one child view for this table then it is a partitioned table.
+            boolean isPartitioned = ViewUtil.hasChildViews(
+                    childLinkHTable,
+                    EMPTY_BYTE_ARRAY,
+                    Bytes.toBytes(schemaName),
+                    Bytes.toBytes(tableName),
+                    currentTime);
+
+            return isPartitioned ?
+                    new PartitionedTableTTLTracker(baseTable, env, isSalted, isPartitionedIndexTable) :
```

Comment:
View indexes are partitioned table but they do not have any children in the child link table. This code does not handle them properly, does it?

Reply:
>> View indexes are partitioned table but they do not have any children in the child link table
Correct!, but if there are no views defined in SYSCAT for this table, we do not expect any rows to be in them.